PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -287,20 +303,103 @@ void DefaultBinning()
+  rh_a.fResultsHistogramsFixedLengthBins[AFO_MULTIPLICITY][1] = 0.;
+  rh_a.fResultsHistogramsFixedLengthBins[AFO_MULTIPLICITY][2] = 5000.;
+
+  rh_a.fResultsHistogramsFixedLengthBins[AFO_CENTRALITY][0] = 100;
+  rh_a.fResultsHistogramsFixedLengthBins[AFO_CENTRALITY][1] = 0.;
+  rh_a.fResultsHistogramsFixedLengthBins[AFO_CENTRALITY][2] = 100.;
+
+  rh_a.fResultsHistogramsFixedLengthBins[AFO_PT][0] = cph_a.fParticleHistogramsBins[ePt][0];
+  rh_a.fResultsHistogramsFixedLengthBins[AFO_PT][1] = cph_a.fParticleHistogramsBins[ePt][1];
+  rh_a.fResultsHistogramsFixedLengthBins[AFO_PT][2] = cph_a.fParticleHistogramsBins[ePt][2];
+
+  rh_a.fResultsHistogramsFixedLengthBins[AFO_ETA][0] = cph_a.fParticleHistogramsBins[eEta][0];
+  rh_a.fResultsHistogramsFixedLengthBins[AFO_ETA][1] = cph_a.fParticleHistogramsBins[eEta][1];
+  rh_a.fResultsHistogramsFixedLengthBins[AFO_ETA][2] = cph_a.fParticleHistogramsBins[eEta][2];
+
+  // d) Variable-length binning set via MuPa-Configurables.h:
+  if (cUseVariableLength_mult_bins) {
+    rh_a.fUseResultsHistogramsVariableLengthBins[AFO_MULTIPLICITY] = kTRUE;
+    rh_a.fResultsHistogramsVariableLengthBinsString[AFO_MULTIPLICITY] = cVariableLength_mult_bins;
+    this->CastStringIntoArray(AFO_MULTIPLICITY);
+  }
+  if (cUseVariableLength_cent_bins) {
+    rh_a.fUseResultsHistogramsVariableLengthBins[AFO_CENTRALITY] = kTRUE;
+    rh_a.fResultsHistogramsVariableLengthBinsString[AFO_CENTRALITY] = cVariableLength_cent_bins;
+    this->CastStringIntoArray(AFO_CENTRALITY);
+  }
+  if (cUseVariableLength_pt_bins) {
+    rh_a.fUseResultsHistogramsVariableLengthBins[AFO_PT] = kTRUE;
+    rh_a.fResultsHistogramsVariableLengthBinsString[AFO_PT] = cVariableLength_pt_bins;
+    this->CastStringIntoArray(AFO_PT);
+  }
+  if (cUseVariableLength_eta_bins) {
+    rh_a.fUseResultsHistogramsVariableLengthBins[AFO_ETA] = kTRUE;
+    rh_a.fResultsHistogramsVariableLengthBinsString[AFO_ETA] = cVariableLength_eta_bins;
+    this->CastStringIntoArray(AFO_ETA);
+  }
 
 } // void DefaultBinning()
 
 //============================================================
 
+void CastStringIntoArray(Int_t AFO)
+{
+  // Temporary function, to be removed eventually. Here temporarily I am casting e.g. a string "1.0,2.0,5.0" into corresponding TArrayD.
+  // TBI 20240114 This function is used until I figure out how to pass array directly via configurable.
+
+  if (tc.fVerbose) {
+    LOGF(info, "\033[1;32m%s\033[0m", __PRETTY_FUNCTION__);
+  }
+
+  if (tc.fVerbose) {
+    LOGF(info, "\033[1;32m Casting a string %s into TArrayD .... \033[0m", rh_a.fResultsHistogramsVariableLengthBinsString[AFO].Data());
+  }
+
+  TObjArray* oa = rh_a.fResultsHistogramsVariableLengthBinsString[AFO].Tokenize(",");
+  if (!oa) {
+    LOGF(fatal, "in function \033[1;31m%s at line %d \n fResultsHistogramsVariableLengthBinsString[AFO] = %s\033[0m", __PRETTY_FUNCTION__, __LINE__, rh_a.fResultsHistogramsVariableLengthBinsString[AFO].Data());
+    cout << __LINE__ << endl;
+    exit(1);
```

Comment:
A `return` would be enough because you will never reach here. And it is more aesthetic

Reply:
Thanks for spotting, I forgot to remove this one. And I see now a few more cout's survived porting from AliPhysics, I can fix this in another commit alongside further development, since this is trivial to fix.

Reply:
I will remove also all exit(1) statements, this is leftover from AliPhysics